### PR TITLE
Allow overriding container image

### LINKED ISSUE #42: Customizable busybox image
This actions fails with this if there were too many builds happening recently.

```
429 Too Many Requests - Server message: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading
```

Would like to have an option to override the busybox image and where it pulls it from, ex: a hosted proxy.

### PR DESCRIPTION
**Pull request by relu**
Fixes #42

Not sure if the built dist files should be included in the PR or not. Let me know and I can remove them if needed.

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -1,399 +1,1656 @@
-lockfileVersion: '6.0'
+lockfileVersion: '9.0'
```
Comment (AkihiroSuda) [inline pnpm-lock.yaml:1]: Probably this should be a separate PR?

File: src/opts.ts [hunk 2]
```diff
@@ -49,6 +51,7 @@ Options:
   --cache-map    The map of actions source paths to container destination paths or mount arguments
   --scratch-dir  Where the action is stores some temporary files for its processing. Default: 'scratch'
   --skip-extraction  Skip the extraction of the cache from the docker container
+  --container-image  The container image to use for injecting and extracting the cache. Default: 'busybox:1'
```
Comment (AkihiroSuda) [inline src/opts.ts:54]: What about calling this `--utility-image`?

**Comment by AkihiroSuda** [comment]
> Fixes #42


Alternatively we may consider just using https://github.com/containerd/containerd/pkgs/container/busybox to avoid the Docker Hub rate limit?

File: pnpm-lock.yaml (hunk 1 above)
Comment (relu) [reply pnpm-lock.yaml]: Agreed. I've committed this change by mistake. Thanks!

**Comment by relu** [comment]
Using the GHCR image as a default makes sense. I think it's still valuable to allow overriding the default container image.

File: src/opts.ts (hunk 2 above)
Comment (relu) [reply src/opts.ts]: I don't have a strong preference, though I think "image" without the accompanying "container" could be ambiguous.

**Comment by relu** [comment]
@AkihiroSuda I just got back to this and noticed that the `ghcr.io/containerd/busybox` repository is no longer actively maintained. The last update there is from two years ago. I've been trying to find an alternative but there's nothing on GHCR. I suppose the only remaining option is to revert to the DockerHub busybox image.

**Review by AkihiroSuda — APPROVED** [review]
Thanks, the BusyBox image is outdated, but I assume practically it doesn't matter
